File: my-react-app/services/usersService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/users'; // Cambia la URL según la configuración de tu backend

// Obtener todos los usuarios
export const getUsers = async () => {
    try {
        const response = await axios.get(API_URL);
        return response.data;
    } catch (error) {
        console.error('Error fetching users', error);
        throw error;
    }
};

// Crear un nuevo usuario
export const createUser = async (user) => {
    try {
        const response = await axios.post(API_URL, user);
        return response.data;
    } catch (error) {
        console.error('Error creating user', error);
        throw error;
    }
};

// Obtener un usuario por ID
export const getUserById = async (id) => {
    try {
        const response = await axios.get(`${API_URL}/${id}`);
        return response.data;
    } catch (error) {
        console.error('Error fetching user', error);
        throw error;
    }
};

// Actualizar un usuario por ID
export const updateUser = async (id, user) => {
    try {
        const response = await axios.put(`${API_URL}/${id}`, user);
        return response.data;
    } catch (error) {
        console.error('Error updating user', error);
        throw error;
    }
};

// Eliminar un usuario por ID
export const deleteUser = async (id) => {
    try {
        await axios.delete(`${API_URL}/${id}`);
    } catch (error) {
        console.error('Error deleting user', error);
        throw error;
    }
};

File: src/main/java/com/skate/skate/services/UsersService.java
package com.skate.skate.services;

import java.util.List;
import java.util.Optional;


import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import com.skate.skate.models.Users;
import com.skate.skate.repositories.UsersRepository; 


@Service
public class UsersService {
    

    private final UsersRepository usersRepository;


    public UsersService(UsersRepository usersRepository){
        this.usersRepository = usersRepository;
    }


       // Obtener todos los usuarios
    public List<Users> getUsers(){
        return usersRepository.findAll();
        }


        // Agregar un nuevo usuario
     public ResponseEntity<Object> addUsers(Users users){
        usersRepository.save(users);
        return new ResponseEntity<>(users, HttpStatus.CREATED);
        }


         // Obtener un usuario por ID
    public ResponseEntity<Object> getUserById(Integer id) {
        Optional<Users> user = usersRepository.findById(id);
        if (user.isPresent()) {
            return new ResponseEntity<>(user.get(), HttpStatus.OK);
        } else {
            return new ResponseEntity<>("Usuario no encontrado", HttpStatus.NOT_FOUND);
        }
    }

    // Actualizar un usuario existente
    public ResponseEntity<Object> updateUser(Integer id, Users updatedUser) {
        Optional<Users> existingUser = usersRepository.findById(id);
        if (existingUser.isPresent()) {
            updatedUser.setId(id); // Asegurarse de que el ID no cambie
            usersRepository.save(updatedUser);
            return new ResponseEntity<>(updatedUser, HttpStatus.OK);
        } else {
            return new ResponseEntity<>("Usuario no encontrado", HttpStatus.NOT_FOUND);
        }
    }

    // Eliminar un usuario
    public ResponseEntity<Object> deleteUser(Integer id) {
        Optional<Users> user = usersRepository.findById(id);
        if (user.isPresent()) {
            usersRepository.deleteById(id);
            return new ResponseEntity<>("Usuario eliminado con éxito", HttpStatus.OK);
        } else {
            return new ResponseEntity<>("Usuario no encontrado", HttpStatus.NOT_FOUND);
        }
    }
}






